digiroad2-api-oth/src/main/scala/fi/liikennevirasto/digiroad2/ChangeApi.scala

```diff
@@ -513,4 +523,197 @@ class ChangeApi(val swagger: Swagger) extends ScalatraServlet with JacksonJsonSu
+
+      Map(
+        "roadNumber" -> roadLink.attributes.getOrElse("VIITE_ROAD_NUMBER", roadLink.attributes.get("TEMP_ROAD_NUMBER")),
+        "roadPart" -> roadLink.attributes.getOrElse("VIITE_ROAD_PART_NUMBER", roadLink.attributes.get("TEMP_ROAD_PART_NUMBER")),
+        "roadTrack" -> roadLink.attributes.getOrElse("VIITE_TRACK", roadLink.attributes.get("TEMP_TRACK")),
+        "roadStartAddr" -> laneStartAddr.toInt,
+        "roadEndAddr" -> laneEndAddr.toInt
+      )
+    }
+
+    //get address and measures of the different segment between the lanes
+    def getDifferentSegmentAddressesAndMeasures(laneChange: LaneChange) = {
+      val roadLink = laneChange.roadLink.get
+      val lane = laneChange.lane
+      val oldLane = laneChange.oldLane.get
+
+      val roadStartAddr = roadLink.attributes.getOrElse("VIITE_START_ADDR", roadLink.attributes.get("TEMP_START_ADDR")).toString.toDouble
+      val roadEndAddr = roadLink.attributes.getOrElse("VIITE_END_ADDR", roadLink.attributes.get("TEMP_END_ADDR")).toString.toDouble
+
+      val laneStartAddr = (roadStartAddr + lane.startMeasure).toInt
+      val laneEndAddr = (roadEndAddr - (roadLink.length - lane.endMeasure)).toInt
+
+      val oldLaneStartAddr = (roadStartAddr + oldLane.startMeasure).toInt
+      val oldLaneEndAddr = (roadEndAddr - (roadLink.length - oldLane.endMeasure)).toInt
+
+      if(laneStartAddr > oldLaneStartAddr){
+        ((oldLaneStartAddr, oldLane.startMeasure), (laneStartAddr, lane.startMeasure))
+      } else if(oldLaneStartAddr > laneStartAddr){
+        ((laneStartAddr, lane.startMeasure), (oldLaneStartAddr, oldLane.startMeasure))
+      } else if(laneEndAddr > oldLaneEndAddr){
+        ((oldLaneEndAddr, oldLane.endMeasure), (laneEndAddr, lane.endMeasure))
+      }else{
+        ((laneEndAddr, lane.endMeasure), (oldLaneEndAddr, oldLane.endMeasure))
+      }
+    }
+
+    Map("type" -> "FeatureCollection",
+      "features" -> laneChanges.flatMap{ laneChange =>
+        val lane = laneChange.lane
+
+        val startDateValue = laneService.getPropertyValue(lane, "start_date")
+        val startDate = if(Try(startDateValue.asInstanceOf[LaneProperty]).isFailure) startDateValue else ""
+
+        val endDateValue = laneService.getPropertyValue(lane, "end_date")
+        val endDate = if(Try(endDateValue.asInstanceOf[LaneProperty]).isFailure) endDateValue else ""
+
+        val laneType = laneService.getPropertyValue(lane, "lane_type")
+
+        val commonJson = Map("type" -> "Feature",
+          "changeType" -> {if(laneChange.changeType == LaneChangeType.Add) "Add" else "Modify"},
+          "geometry" -> getGeometryMap(laneChange.roadLink.get),
+          "startDate" -> startDate,
+          "endDate" -> endDate,
+          "linkId" -> lane.linkId,
+          "startMeasure" -> lane.startMeasure,
+          "endMeasure" -> lane.endMeasure) ++ mapLaneAddressInfo(lane, laneChange.roadLink.get)
+
+        val json = laneChange.changeType match {
```

Comment:
The bellow the
laneChange.changeType match

Is 90% to decide which create and modify fields will be used.
Shouldn't we create a function to decide which value should be return and have it in a more generic map  like commonJson ?

Even for Lengthened and Shortened that have more logic we can use for those too.

Reply:
Done